src/Microsoft.DotNet.Darc/DarcLib/Helpers/GitRepoUrlParser.cs

```diff
@@ -2,15 +2,62 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
 
 #nullable enable
 namespace Microsoft.DotNet.DarcLib.Helpers;
 
 public static class GitRepoUrlParser
 {
+    public static GitRepoType ParseTypeFromUri(string pathOrUri)
+    {
+        if (!Uri.TryCreate(pathOrUri, UriKind.Absolute, out Uri? parsedUri))
+        {
+            return GitRepoType.None;
+        }
+
+        return parsedUri switch
+        {
+            { IsFile: true } => GitRepoType.Local,
+            { Host: "github.com" } => GitRepoType.GitHub,
+            { Host: var host } when host is "dev.azure.com" => GitRepoType.AzureDevOps,
+            { Host: var host } when host.EndsWith("visualstudio.com") => GitRepoType.AzureDevOps,
+            _ => GitRepoType.None,
+        };
+    }
+
+    /// <summary>
+    /// Sorts so that we go Local -> GitHub -> AzDO.
+    /// (in other words local, public, internal)
```

Comment:
Nit: the doc does not fully match the method name in repo type naming (`Other` vs `internal / AzDO`)